doc/source/contribution_and_api/unit_testing.rst

```diff
@@ -0,0 +1,84 @@
+.. _ref_unit_testing_contributing:
+
+Unit Testing
+============
+
+Unit tests are testing functions which validate that the software
+logic implemented inside a certain method, class or module is
+working as expected. They should try to be as atomic and 
+independent as possible.
+
+Unit testing is highly important. It enables to check that the
+changes on the code are consistent with other parts of the code
+and verify that the changes are the expected in the implementation.
+
+Unit tests are in the ``tests`` directory inside this repository.
+All the tests that you will find in this  directory are not
+unit tests. Some of them are integration tests. The difference between
+a unit test and an integration one is that integration tests will
+test several units of the code and ensure that they work together.
+
+To verify that all the code is properly tested, `Codecov <https://github.com/codecov>`_
+generates a coverage report of the committed code. It details how
+merging the Pull Request will impact the coverage. It is one of
+the checks that must be approved to merge the code.
+
```

Comment:
I think it will be nice if you can add one example of coverage.

Check: https://www.atlassian.com/continuous-delivery/software-testing/code-coverage